backend/src/signer/signer.rs

```diff
@@ -0,0 +1,125 @@
+use crate::contracts::generated::summa_contract::Summa;
+use ethers::{
+    abi::{encode, Token},
+    prelude::SignerMiddleware,
+    providers::{Http, Middleware, Provider},
+    signers::{LocalWallet, Signer, WalletError},
+    types::{Address, Signature},
+    utils::keccak256,
+};
+use futures::future::join_all;
+use serde_json::Value;
+use std::{error::Error, fs::File, io::BufReader, path::Path, str::FromStr, sync::Arc};
+
+#[derive(Debug)]
+pub struct SummaSigner {
+    signing_wallets: Vec<LocalWallet>,
+    summa_contract: Summa<SignerMiddleware<Provider<Http>, LocalWallet>>,
+}
+
+impl SummaSigner {
+    pub fn new(
+        private_keys: &[&str],
+        main_signer_key: &str,
+        chain_id: u64,
+        rpc_url: &str,
+        address: Address,
+    ) -> Self {
+        let wallet: LocalWallet = LocalWallet::from_str(main_signer_key).unwrap();
+
+        let provider = Provider::<Http>::try_from(rpc_url).unwrap();
+        let client = Arc::new(SignerMiddleware::new(
+            provider,
+            wallet.with_chain_id(chain_id),
+        ));
+
+        // An example of how to retrieve the contract address from the deployments.json file
```

Comment:
Regarding my previous comment, I believe this is a good example to place in the `example` folder. 
For now, we can just leave it as is.